### ISSUE TITLE
Performance regression in 0.7.53

### ISSUE THREAD

**Issue by mmd-osm** [drolbr/Overpass-API#323]
There's some new performance regression with the following query in 0.7.53:

```
node(51.363934891891894,5.170799818181818,51.580151108108105,5.534436181818181)["rcn_ref"];out;
```

Affected are dev:/api_drolbr, German, French instance (15s+ runtime)
Not Affected: Rambler, still on 0.7.52 (2,5s runtime), and [my 0.7.53++ instance](http://www.openstreetmap.org/user/mmd/diary/39568) (800ms runtime)

Also related: https://wiki.openstreetmap.org/wiki/Overpass_API/status, https://help.openstreetmap.org/questions/52384/very-restrictive-usage-limit-on-the-overpass-api-since-a-while

**Comment by mmd-osm** [comment]
OK, I think it's pretty clear now: it's a performance regression which has already been fixed in #301, but that pull request hasn't been merged yet!

```
+  67.12%  osm3s_query  osm3s_query          [.] void std::__adjust_heap<__gnu_cxx::__normal_iterator<Uint64*, std::vector<Uint64, std::allocator<Uint64> > >, long, Uint64>◆
+  24.25%  osm3s_query  osm3s_query          [.] void std::__introsort_loop<__gnu_cxx::__normal_iterator<Uint64*, std::vector<Uint64, std::allocator<Uint64> > >, long>(__gn▒
+   3.02%  osm3s_query  osm3s_query          [.] void generate_ids_by_coarse<Uint32_Index, Node_Skeleton>(std::set<Uint32_Index, std::less<Uint32_Index>, std::allocator<Uin▒
+   1.20%  osm3s_query  libz.so.1.2.8        [.] inflate_fast                                                                                                               ▒
+   0.99%  osm3s_query  osm3s_query          [.] __gnu_cxx::__normal_iterator<Uint64*, std::vector<Uint64, std::allocator<Uint64> > > std::unique<__gnu_cxx::__normal_iterat▒
+   0.68%  osm3s_query  libc-2.19.so         [.] 0x000000000008077c                                                                                                         ▒
+   0.40%  osm3s_query  osm3s_query          [.] void collect_items_range<Uint32_Index, Node_Skeleton, std::set<std::pair<Uint32_Index, Uint32_Index>, std::less<std::pair<U▒
+   0.25%  osm3s_query  osm3s_query          [.] std::map<unsigned int, std::vector<Uint64, std::allocator<Uint64> >, std::less<unsigned int>, std::allocator<std::pair<unsi▒
+   0.19%  osm3s_query  osm3s_query          [.] bool std::binary_search<__gnu_cxx::__normal_iterator<Uint64*, std::vector<Uint64, std::allocator<Uint64> > >, Uint64>(__gnu▒
+   0.17%  osm3s_query  libz.so.1.2.8        [.] adler32                                                                                                                    ▒
+   0.17%  osm3s_query  libc-2.19.so         [.] malloc                                                                                                                     ▒
+   0.13%  osm3s_query  osm3s_query          [.] Bbox_Constraint::filter(Resource_Manager&, Set&, unsigned long long)                                                       ▒
+   0.07%  osm3s_query  osm3s_query          [.] Handle<Uint64>::object() const                                                                                             ▒
+   0.06%  osm3s_query  libstdc++.so.6.0.19  [.] operator new(unsigned long)                                                                                                ▒
+   0.05%  osm3s_query  osm3s_query          [.] std::vector<Node_Skeleton, std::allocator<Node_Skeleton> >::_M_insert_aux(__gnu_cxx::__normal_iterator<Node_Skeleton*, std:▒
+   0.05%  osm3s_query  libz.so.1.2.8        [.] inflate_table                                                                                                              ▒
+   0.05%  osm3s_query  [kernel.kallsyms]    [k] page_fault                                                                                                                 ▒
+   0.05%  osm3s_query  osm3s_query          [.] Random_File_Index::Random_File_Index(File_Properties const&, bool, bool, std::string const&, std::string const&)           ▒
+   0.05%  osm3s_query  [kernel.kallsyms]    [k] copy_user_enhanced_fast_string                                                                                             ▒
+   0.04%  osm3s_query  osm3s_query          [.] Block_Backend_Basic_Iterator<Tag_Index_Local, Uint64>::index() [clone .isra.361]                                           ▒
+   0.04%  osm3s_query  osm3s_query          [.] Block_Backend_Range_Iterator<Tag_Index_Local, Uint64, std::_Rb_tree_const_iterator<Tag_Index_Local> >::operator++()        ▒
+   0.04%  osm3s_query  osm3s_query          [.] void Query_Statement::filter_by_tags<Uint32_Index, Node_Skeleton>(std::map<Uint32_Index, std::vector<Node_Skeleton, std::al▒
+   0.04%  osm3s_query  osm3s_query          [.] void filter_ids_by_tags<Uint64>(std::map<std::string, std::pair<std::string, std::vector<Regular_Expression*, std::allocato▒
+   0.04%  osm3s_query  libc-2.19.so         [.] free                                                                                                                       ▒
+   0.03%  osm3s_query  osm3s_query          [.] void collect_tags_framed<Uint64>(std::map<Uint64, std::vector<std::pair<std::string, std::string>, std::allocator<std::pair▒
+   0.03%  osm3s_query  libstdc++.so.6.0.19  [.] std::string::assign(std::string const&)                                                                                    ▒
+   0.03%  osm3s_query  osm3s_query          [.] File_Blocks_Index<Uint31_Index>::File_Blocks_Index(File_Properties const&, bool, bool, std::string const&, std::string cons▒
+   0.03%  osm3s_query  [kernel.kallsyms]    [k] clear_page_c_e                                                                                                             ▒
+   0.03%  osm3s_query  osm3s_query          [.] Handle<Node_Skeleton>::update_ptr() const                                                                                  ▒
+   0.03%  osm3s_query  osm3s_query          [.] void std::sort<__gnu_cxx::__normal_iterator<Uint64*, std::vector<Uint64, std::allocator<Uint64> > > >(__gnu_cxx::__normal_i▒
+   0.03%  osm3s_query  osm3s_query          [.] Block_Backend_Basic_Iterator<Uint32_Index, Node_Skeleton>::advance()                                                       ▒
+   0.03%  osm3s_query  libstdc++.so.6.0.19  [.] std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&)                   ▒
+   0.03%  osm3s_query  [kernel.kallsyms]    [k] __mem_cgroup_commit_charge                                                                                                 ▒
+   0.03%  osm3s_query  [kernel.kallsyms]    [k] mem_cgroup_charge_common                                                                                                   ▒
+   0.02%  osm3s_query  osm3s_query          [.] Tag_Index_Local::Tag_Index_Local(void*)                                                                                    ▒
+   0.02%  osm3s_query  osm3s_query          [.] Block_Backend_Range_Iterator<Uint32_Index, Node_Skeleton, std::_Rb_tree_const_iterator<std::pair<Uint32_Index, Uint32_Index▒
+   0.02%  osm3s_query  libstdc++.so.6.0.19  [.] free@plt                                                                                                                   ▒
+   0.02%  osm3s_query  libstdc++.so.6.0.19  [.] malloc@plt                                                                                                                 ▒
+   0.02%  osm3s_query  libz.so.1.2.8        [.] inflate                                                                                                                    ▒
+   0.02%  osm3s_query  osm3s_query          [.] Handle<Node_Skeleton>::object() const                                                                                      ▒
```

**Comment by mmd-osm** [comment]
Closing, as issue was fixed in c30c154df701d20cf1ec3e9274910d9dce31ce7d..e6394bff10c7ea8a9c68c876854b4b13feb85a3d

Thanks!
